config/cmake/ConfigureChecks.cmake

```diff
@@ -64,7 +64,7 @@ if (WIN32 AND NOT MINGW)
 endif ()
 
 if (WINDOWS)
-  set (HDF4_REQUIRED_LIBRARIES "ws2_32.lib;wsock32.lib")
+  set (CMAKE_REQUIRED_LIBRARIES "ws2_32.lib;wsock32.lib")
```

Comment:
Why the change?

Reply:
I was trying to get MinGW to work and this seemed like the more correct thing to do. Is it not?

Reply:
Isn't that the point, though? We want to require links to ws2_32, etc.

Reply:
Actually it looks like the opposite is true -  CMAKE_REQUIRED_LIBRARIES is temporary for checking, we use HDF4_REQUIRED_LIBRARIES after the configure_checks.

Reply:
But then again CMAKE_REQUIRED_LIBRARIES shows up afterwards - ARRGH.

Reply:
HDF5 uses CMAKE_REQUIRED_LIBRARIES is temporary for checking.

Reply:
For HDF4 CMakeLists.txt line 485 incorrectly set CMAKE_REQUIRED_LIBRARIES and should have been setting HDF4_REQUIRED_LIBRARIES.
Then everything would be correct ;
CMAKE_REQUIRED_LIBRARIES is temporary for checking, we use HDF4_REQUIRED_LIBRARIES after the configure_checks.